terraform/deployments/cluster-services/main.tf

```
@@ -1,23 +1,38 @@
-# The eks-stage2 module is responsible for Kubernetes objects within the EKS
-# cluster.
+# The cluster-services module is responsible for Kubernetes objects within the
+# EKS cluster.
 #
-# It has to be a separate root module (aka deployment/project) because
-# Terraform does not handle dependencies in provider configurations, which
-# means the Kubernetes and Helm providers cannot reliably be initialised in the
-# same root module which creates the EKS cluster (see warning in
-# https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#stacking-with-managed-kubernetes-cluster-resources).
+# Any AWS resources relating to the cluster belong in
+# ../cluster-infrastructure, not in this module.
+#
+# See https://github.com/alphagov/govuk-infrastructure/blob/main/docs/architecture/decisions/0003-split-terraform-state-into-separate-aws-cluster-and-kubernetes-resource-phases.md
 
 terraform {
   backend "s3" {}
+}
 
-  required_providers {
-    aws = {
-      source  = "hashicorp/aws"
-      version = "~> 3.33"
-    }
+provider "kubernetes" {
+  host                   = data.terraform_remote_state.cluster_infrastructure.outputs.cluster_endpoint
+  cluster_ca_certificate = base64decode(data.terraform_remote_state.cluster_infrastructure.outputs.cluster_certificate_authority_data)
+  exec {
+    # aws eks get-token (at least up to aws-cli 2.2.30) still uses the v1alpha1 API.
+    # Amazon fluffed the migration to v1beta1 and had to revert it:
+    # https://github.com/aws/aws-cli/pull/6309
```

Nit, I wonder if this comment could be rewritten as a TODO? Perhaps

```suggestion
    # TODO: use v1beta1 API once `aws eks get-token` supports this
    # https://awscli.amazonaws.com/v2/documentation/api/latest/reference/eks/get-token.html
```

---

Yeah that's a good shout, thanks. Done.